app/src/functions/send-discord-notification/app.ts

```diff
@@ -0,0 +1,106 @@
+import { getQueryResults } from '../../lib/datadog';
+import { ApplicationLog, isApplicationLog, LambdaLog, LoadBalancerLog } from '../../models/datadog';
+import { getHourFromUtcDate, getTodayStringTime } from '../../lib/date-utils';
+import { formatNumber } from '../../utils/helpers';
+import { sendEmbedMessage, sendMessage } from '../../lib/discord';
+import { EmbedFieldData } from 'discord.js';
+
+type availableLogs = LoadBalancerLog | ApplicationLog
+
+// Goes through all the values and constantly updates the map with the most recent one,
+// as the logs already come sorted, thereby only keeping the latest log of each hour
+function filterMinimunDuplicates<T extends LambdaLog>(logs: T[]): T[] {
+  const filtered: T[] = []
+
+  const filter = new Map<string, T>()
+  logs.forEach(lb => filter.set(lb.hourstamp, lb))
+
+  for (const [_, lb] of filter.entries()) {
+    filtered.push(lb)
+  }
+
+  return filtered
+}
+
+// Returns a map of exceeded logs based on their hourstamp 
+function mapsExceededThresholds(logs: availableLogs[]): Map<string, availableLogs[]> {
```

Comment:
I'm being a bit bikesheddy here, but 
```suggestion
function mapExceededThresholds(logs: availableLogs[]): Map<string, availableLogs[]> {
```